engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ArrowFixedArrayDate.java

```diff
@@ -222,37 +162,11 @@ private static ZonedDateTime zonedDateTimeFromSeconds(long seconds, long nano, Z
     return Instant.ofEpochSecond(seconds, nano).atZone(zone);
   }
 
-  @CompilerDirectives.TruffleBoundary
-  private static Instant instantForZone(
-      LocalDate date, LocalTime time, ZoneId zone, ZoneId target) {
-    return date.atTime(time).atZone(zone).withZoneSameLocal(target).toInstant();
-  }
-
-  @CompilerDirectives.TruffleBoundary
-  private static Instant instantForOffset(LocalDate date, LocalTime time, ZoneOffset offset) {
-    return date.atTime(time).toInstant(offset);
-  }
-
-  public enum DateUnit implements SizeInBytes {
-    Day(4),
-    Millisecond(8);
-
-    private final int bytes;
-
-    DateUnit(int bytes) {
-      this.bytes = bytes;
-    }
-
-    public int sizeInBytes() {
-      return bytes;
-    }
-  }
-
-  private static final long nanoDiv = 1000000000L;
+  public static final long NANO_DIV = 1000000000L;
 
-  private static final ZoneId utc = ZoneId.of("UTC");
+  public static final ZoneId UTC = ZoneId.of("UTC");
 
-  private static int typeAdjustedIndex(long index, SizeInBytes unit) {
-    return Math.toIntExact(index * unit.sizeInBytes());
+  public static int typeAdjustedIndex(long index, int daySizeInBytes) {
```

Comment:
Package private might be enough. Whole language seems to be in a single package (except `ArrowParser`).